bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BallerinaXmlSerializer.java

```diff
@@ -137,10 +137,42 @@ private void writeXMLComment(XmlComment xmlValue) throws XMLStreamException {
     }
 
     private void writeXMLText(XmlText xmlValue) throws XMLStreamException {
-        String textValue = xmlValue.getTextValue();
-        if (!textValue.isEmpty()) {
-            xmlStreamWriter.writeCharacters(textValue);
+        // No need to escape xml text when they are within xml element, it's handle from xml stream writer.
+        if (this.withinElement) {
+            String textValue = xmlValue.getTextValue();
+            if (!textValue.isEmpty()) {
+                xmlStreamWriter.writeCharacters(textValue);
+            }
+        } else {
+            char[] textValue = escapeCharacters(xmlValue.getTextValue());
```

Comment:
Wouldn't it be cleaner to do this in io.ballerina.runtime.internal.values.XmlText#stringValue that is being removed (and maybe even cache the value) instead of changing anything here?

Reply:
That's a workaround to fix the current case, IMO XmlSerializer should be able to serialize anything